### PR TITLE
HPCC-26757 Inconsistencies between systemd file dafilesrv@mydafilesrv.service and dafilesrv program

### PR DESCRIPTION
**Pull request by lpezet**
## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:

Sorry I cheated here a bit. This is actually HPCC-26750, HPCC-26757, and HPCC-26761. I wasn't sure how to go about it in 3 different pull requests.
Long story short, HPCC Platform, for me, was NOT working at all. Those fixes, I believe, seem to make it at least start everything fine (Dali, Dafilesrv, and more importantly Thor Slave).
I have NOT ran any tests, beyond installing the package and going through preflight checks.

### CODE REVIEW

File: dali/server/daserver.cpp [hunk 1]
```diff
@@ -475,6 +475,7 @@ int main(int argc, const char* argv[])
 
         for (unsigned i=1;i<(unsigned)argc;i++) {
             if (streq(argv[i],"--daemon") || streq(argv[i],"-d")) {
+                i++; // swallowing <instanceName> after --daemon|-d
```
Comment (Michael-Gardner) [inline dali/server/daserver.cpp:478]: This will work fine.  Technically this is part of HPCC-26750 but it is up to @richardkchapman how we include it.

You might want to add some error checking for instanceName existing. ex:  i + 1 < argc to ensure the next argument exists, !strncmp(instanceName,"-",1) to check that the value following the daemon flag isn't another flag.

File: initfiles/bash/etc/systemd/system/CMakeLists.txt [hunk 2]
```diff
@@ -32,7 +32,7 @@ set(componentList
     backupnode
     toposerver)
 
-set(dafilesrv "dafilesrv" "-L ${LOG_PATH} -I %i -D" "")
+set(dafilesrv "dafilesrv" "--logDir=${LOG_PATH} --name=%i --daemon" "")
```
Comment (Michael-Gardner) [inline initfiles/bash/etc/systemd/system/CMakeLists.txt:35]: This is correct. This should target 8.4.x I believe. The change to the Usage for dafilesrv happened in https://github.com/hpcc-systems/HPCC-Platform/commit/1bc5da6d3e1e55aa37819b0f904d702c9b501fde#.  

Also note; the --daemon call here does not require an instanceName after it because dafilesrv is using its own code to fork as a daemon, not the code in jutil.

File: initfiles/bash/etc/systemd/system/CMakeLists.txt [hunk 3]
```diff
@@ -42,7 +42,7 @@ set(esp "esp" "--daemon %i" "dafilesrv.service")
 set(roxie "roxie" "--topology=RoxieTopology.xml --logfile --restarts=2 --stdlog=0 --daemon %i" "dafilesrv.service")
 set(sasha "saserver" "--daemon %i" "dafilesrv.service")
 set(thor "thormaster_lcr" "--daemon %i MASTER=\${THORMASTER}:\${THORMASTERPORT}" "dafilesrv.service")
-set(thorslave "thorslave_lcr" "--daemon %i master=\${THORMASTER}:\${THORMASTERPORT} slave=.:\${SLAVEPORT} slavenum=\${SLAVENUM} logDir=${LOG_PATH}/\${THORNAME}" "dafilesrv.service")
+set(thorslave "thorslave_lcr" "--daemon %i --master=\${THORMASTER}:\${THORMASTERPORT} --slave=.:\${SLAVEPORT} --slavenum=\${SLAVENUM} --logDir=${LOG_PATH}/\${THORNAME}" "dafilesrv.service")
```
Comment (Michael-Gardner) [inline initfiles/bash/etc/systemd/system/CMakeLists.txt:45]: These changes could target back to 7.8.x if we wanted to, but I don't think we plan on building an unsupported version.  This happened when we switched over from using mergeCmdParams() for globals, to loadArgsIntoConfiguration().  https://github.com/hpcc-systems/HPCC-Platform/commit/bd9e704545aa29ec05265b4c0363492c7f08ec22#diff-6095418795f3a03d23c0932453ccb5821ce8b11029717e8cdb861924cfe835a5

Tested this in a 7.8 build and it worked properly.

**Review by Michael-Gardner — COMMENTED** [review]
@lpezet thanks for the submission. We might need to slice this commit up into separate commits because the regression/bugs were introduced in different branches.

**Comment by richardkchapman** [comment]
This PR has been superseded by #15535, #15533, #15531
